Assets/Scripts/AvoidantEnemy.cs

```diff
@@ -2,16 +2,18 @@
 using System.Collections.Generic;
 using UnityEngine;
 
-public class AvoidantEnemy : MonoBehaviour
+public class AvoidantEnemy : Enemy
 {
-    // Start is called before the first frame update
 
-    private Transform target;
     public float speed = 5.0f;
     public bool spawned = false;
 
     void Start()
     {
+        MaxHealth = 3;
+        Health = MaxHealth;
+        Strength = 1;
+        Speed = 20;
         target = GameObject.FindGameObjectWithTag("Corner").transform;

```

Comment:
This might be a little problematic when we combine your code with the new Room concept. This would mean we would have to specify a corner for each room. It would also be problematic because there would be no way for a room to keep track of those additionally spawned "fly" enemies, since the new Room uses manually placed spawn locations that spit out enemies at the start of the game. (If you manage to figure this out, more power to ya!)

For simplicity's sake, it might be it's best to stick to normal enemies for the prototype. But yeah, this avoidant enemy is definitely something that would be worth including in the final game (Isaac has them, after all!)

Reply:
Well not every room need have an Avoidant Enemy or Spawner enemy. The spawned enemies are actual enemy clones. They have an Enemy tag and there is a bool value used to track if the "fly" enemies have spawned. I suppose its a matter of adding a condition check in the Room script to test for this bool value, and then we need to figure out how to account for these "fly" enemies within a particular room.